platform-sdk/docs/proposals/TSS-Ledger-Id/TSS-Ledger-Id.md

```diff
@@ -241,20 +269,159 @@ process for a node and the process of changing the consensus roster.
+
+1. KeyType - An enum that is one of TSS_MESSAGE or TSS_VOTE.
+2. RosterHash - The hash of the consensus roster that the message or vote is related to.
+3. SequenceNumber - The order in which the message or vote came to consensus.
+
+The value associated with a TSS_MESSAGE key is the pair (ShareId, TssMessage)
+
+1. ShareId - The id of the share that the message is related to.
+2. TssMessage - the raw bytes of the TssMessage
+
+The value associated with a TSS_VOTE key is a bit vector with the following interpretation
+
+* The order of bits from least significant bit to most significant bit corresponds to the numeric order of share ids
+  from least to greatest.
+* If a bit is set to 1, then the TssMessage for the corresponding ShareId was valid and contributed to a successful
+  reconstruction of the ledger id.
+* If a bit is set to 0, then the TssMessage for the corresponding ShareId was either invalid, was not received, or
+  was not used in the reconstruction of the ledger id.
+
+Lifecycle Invariants
+
+1. Inserts should only be made for candidate rosters that are going through the keying process.
+2. When a roster is no longer stored in the state, the corresponding key material should be removed.
+3. Every roster that has become active after the TSS Genesis process must have sufficient key material to be
+   able to recover the ledger id.
+
+#### New Wiring Components
+
+The following are new components in the system:
+
+1. TSS State Manager (Not Wired)
+2. TSS Message Creator (Wired)
+3. TSS Message Validator (Wired)
+4. TSS Key Manager (Wired)
+
+##### TSS State Manager
+
+The TSS state manager is executed on the transaction handling thread and is capable of reading and writing the state
+during round handling. The logic will be encapsulated in a class with a well-defined API, but this will not be a fully
+fledged wiring component outside the consensus round handler.
+
+Responsibilities:
+
+1. Handle `TssMessage` system transactions
+    - handled in consensus order
+    - insert into the TSS data map if it is legal to do so
+        - don’t insert multiple messages for the same node
+        - don’t insert any messages if voting window has closed
+    - if inserted into the TSS data map, ensure that the message is forwarded to the TSS Message Validator
+2. Handle `TssVote` system transactions
+    - handled in consensus order
+    - insert into TSS data map if it is legal to do so
+        - don’t insert multiple votes for the same node
+        - don’t insert once voting is closed
+    - after insertion, re-tally votes and decide if voting is now closed
+    - if voting is closed as the result of a new vote,
+        - if the “no” votes win, then the candidate roster will never be adopted
+        - if the “yes” votes win, the candidate roster is now officially confirmed, and will be adopted at the next
+          upgrade boundary
+3. Detect when a new candidate roster is set.
+    - update the pending roster hash in the platform state
```

Comment:
Per the current `TSS-Roster` proposal, all rosters (active, candidate, previous, etc.) will be stored in a map in the state, and the app would indicate that there's a new candidate roster by setting a candidate roster hash in the state somewhere (TBD where.) But the point is that this particular action described in this line would've already been performed by the app. The "TSS State Manager" would only need to detect that this hash has been written to this place and the proceed with informing other components as described just below.

Reply:
removed the line that is the responsibility of the TSS-Roster effort.